app/models/user.rb

```diff
@@ -174,4 +176,10 @@ def activate!
   def deactivate!
     update! active: false
   end
+
+  def check_user_role_provider
+    return unless role
+
+    errors.add(:user_provider, 'has to be the same as the Role provider') if provider != role.provider
+  end
```

Comment:
I think this is too broad of a check. Can we move this check to `users#update` instead?

Reply:
What do you mean by too broad?
If we add the logic to #update instead, shouldn't we add it to #create as well?
IMO this is redundant, we have specs for that already, it shouldn't happen. 
I feel like this belongs in a Validation.

Reply:
Yeah but this check will run on every change to user. We might want to change if to run only if the role is changing, or only do the check in the controller instead